src/librustc_mir/interpret/place.rs

```diff
@@ -348,8 +348,13 @@ where
                 offsets[usize::try_from(field).unwrap()],
             layout::FieldPlacement::Array { stride, .. } => {
                 let len = base.len(self)?;
-                assert!(field < len, "Tried to access element {} of array/slice with length {}",
-                    field, len);
+                if field >= len {
+                    debug!("Tried to access element {} of array/slice with length {}", field, len);
+                    return Err(EvalError {
+                        kind: InterpError::BoundsCheck { len, index: field },
+                        backtrace: None,
+                    });
```

Comment:
Please use the `err!` macro.

@oli-obk looks like I forgot to make the `backtrace` field private. :/

Reply:
done